### PR TITLE
Add-user-api

### PR DESCRIPTION
**Pull request by hazeleclipse**
Add in basic CRUD functionality for the user resource.

### CODE REVIEW

**Comment by hazeleclipse** [comment]
Possible changes
-use username instead of email to identify user
----make sure username is unique

Possible add ins
-Need to add in file/class/function comments

**Review by hazeleclipse — COMMENTED** [review]
Test Comment

File: controllers/userController.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    } catch (err) {
+        // Configure response to failed request
+        res.status(400).json({
+            status: 'fail',
+            message: err
+        });
+    }
+};
+
+exports.getAllUsers = async (req, res) => {
```
Comment (knippel-dev) [inline controllers/userController.js]: Unimpliment get all users as this is a security risk.

Comment (knippel-dev) [reply controllers/userController.js]: removed

**Pull request merged**